src/python/WMCore/MicroService/Unified/Transferor.py

```diff
@@ -72,88 +86,290 @@ def close(self):
+                self.logger.debug("### nodes %s %s" % (nodes, rNodes))
                 subset = set(nodes) & set(rNodes)
                 if subset == set(rNodes):
                     sdict[dataset] = 1
                 else:
                     pct = float(len(subset))/float(len(set(rNodes)))
                     sdict[dataset] = pct
-        if self.verbose:
-            print("### sdict", sdict)
+        self.logger.debug("### sdict %s" % sdict)
         tot = len(sdict.keys())
         if not tot:
             return -1
         # return percentage of completion
         return round(float(sum(sdict.values()))/float(tot), 2) * 100
 
+    def checkTransferRequests(self, request):
+        "Check all transfer requests for given request"
+        # put logic to find out all transfer requests
+        return True
+
     def checkStatus(self, request):
         "Check status of request in local storage"
-        if self.verbose:
-            print("### checkStatus of request: %s" % request)
+        self.logger.debug("### checkStatus of request: %s" % request)
         if self.store.exists(request):
-            completed = self.checkPhedex(request)
+
+            # check subscription status of the request
+            # completed = self.checkSubscription(request)
+            completed = 100
             if completed == 100: # all data are staged
-                if self.verbose:
-                    print("### request is completed, change its status and remove it from the store")
-                # call ReqMgr2 API to change status of the request
-                # self.reqmgr.changeStatus(status)
-                self.store.delete(request)
+                self.logger.debug("### request is completed, change its status and remove it from the store")
+                self.change(request, 'staged', '### transferor')
             else:
-                if self.verbose:
-                    print("### request %s, completed %s" % (request, completed))
-                self.store.update(request, {'PhedexStatus': completed})
+                self.logger.debug("### request %s, completed %s" % (request, completed))
+                self.store.update(request, {'status': completed})
+
+    def change(self, request, req_status, prefix='###'):
+        """
+        Change request status, internally it is done via PUT request to ReqMgr2:
+        curl -X PUT -H "Content-Type: application/json" \
+             -d '{"RequestStatus":"staging", "RequestName":"bla-bla"}' \
+             https://xxx.yyy.zz/reqmgr2/data/request
+        """
+        try:
+            rec = self.store.get(request)
+            self.logger.debug('%s changing %s status of record %s to req_status=%s' \
+                % (prefix, request, rec, req_status))
+            if rec['reqStatus'] != req_status:
+                self.svc.reqmgr.updateRequestStatus(request, req_status)
+                self.store.update(request, {'reqStatus': req_status})
```

Comment:
I'd rather not save the request status in multiple databases. We know sooner or later it goes out of sync and it just gives us more problems. I'd say we don't store request status in this self.store temp db.

Reply:
I agree, but since there is no yet couchdb one I used internal store to test whole system. As I wrote in previous comments it is most likely will be gone once we'll have couchdb one.